app/src/main/java/com/goldenraven/devkitwallet/ui/ElectrumScreen.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2020-2022 thunderbiscuit and contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the ./LICENSE file.
+ */
+
+package com.goldenraven.devkitwallet.ui
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.material3.Button
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.Switch
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.goldenraven.devkitwallet.data.Wallet
+import com.goldenraven.devkitwallet.ui.theme.DevkitWalletColors
+import com.goldenraven.devkitwallet.ui.theme.firaMono
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun ElectrumScreen(navController: NavController) {
+    val focusManager = LocalFocusManager.current
+    val isBlockChainCreated = Wallet.isBlockChainCreated()
+    val electrumServer: MutableState<String> = remember { mutableStateOf("") }
+    val isChecked: MutableState<Boolean> = remember { mutableStateOf(false) }
+    if (isBlockChainCreated) {
+        electrumServer.value = Wallet.getElectrumURL()
+        isChecked.value = Wallet.getElectrumSettings()
+    }
+
+    Scaffold(
+        topBar = { AwayFromHomeAppBar(navController, "Electrum Server") },
+        containerColor = DevkitWalletColors.night4
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(all = 32.dp),
+        ) {
+
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Text(
+                    text = "Use default electrum URL",
+                    color = DevkitWalletColors.snow1,
+                    fontSize = 14.sp,
+                    fontFamily = firaMono,
+                    textAlign = TextAlign.Center,
+                )
+                Switch(
+                    checked = isChecked.value,
+                    onCheckedChange = {
+                        isChecked.value = it
+                        if (it) Wallet.setElectrumSettings(true) else Wallet.setElectrumSettings(false)
+                    },
+                    enabled = isBlockChainCreated
+                )
+            }
+
+            OutlinedTextField(
```

Comment:
Let's disable the OutlinedTextField if the user has the toggle for `useDefaultElectrum` to _on_. That way the text can only be entered if the toggle is set to use a custom Electrum server.